src/index.ts

```diff
@@ -110,60 +188,208 @@ const writeFile = (
-        build_number: condaPackageInfo.build_number
-      };
-
-      if (!FS.analyzePath(`${condaMetaDir}`).exists) {
-        FS.mkdirTree(`${condaMetaDir}`);
-      }
+const initPrimaryPhase = async (
+  pythonPackage: IEmpackEnvMetaPkg,
+  pythonVersion: number[],
+  verbose: boolean,
+  untarjs: IUnpackJSAPI,
+  Module: any,
+  pkgRootUrl: string,
+  prefix: string
+) => {
+  let url = pythonPackage.url
+    ? pythonPackage.url
+    : `${pkgRootUrl}/${pythonPackage.filename}`;
+  if (verbose) {
+    console.log(`Installing a python package from ${url}`);
+  }
+  await installCondaPackage(prefix, url, Module.FS, untarjs, verbose);
+  await Module.init_phase_1(prefix, pythonVersion, verbose);
+};
 
-      if (verbose) {
-        console.log(
-          `Creating conda-meta file for ${condaPackageInfo.name}-${condaPackageInfo.version}-${condaPackageInfo.build} package`
+export const bootstrapFromEmpackPackedEnvironment = async (
+  packagesJsonUrl: string,
+  verbose: boolean = true,
+  skipLoadingSharedLibs: boolean = false,
+  Module: any,
+  pkgRootUrl: string
+): Promise<IPackagesInfo> => {
+  if (verbose) {
+    console.log('fetching packages.json from', packagesJsonUrl);
+  }
+
+  let empackEnvMeta = await fetchJson(packagesJsonUrl);
+  let allPackages: IEmpackEnvMetaPkg[] = empackEnvMeta.packages;
+  let prefix = empackEnvMeta.prefix;
+  let { pythonPackage, pythonVersion, packages } = splitPackages(allPackages);
+  let packagesData = { prefix, pythonVersion };
+
+  if (verbose) {
+    console.log('installCondaPackage');
+  }
+  const untarjsReady = initUntarJS();
+  const untarjs = await untarjsReady;
+
+  if (pythonPackage && pythonVersion) {
+    await initPrimaryPhase(
+      pythonPackage,
+      pythonVersion,
+      verbose,
+      untarjs,
+      Module,
+      pkgRootUrl,
+      prefix
+    );
+  }
+  if (packages?.length) {
+    let sharedLibs = await Promise.all(
+      packages.map(pkg => {
+        const packageUrl = pkg?.url ?? `${pkgRootUrl}/${pkg.filename}`;
+        if (verbose) {
+          console.log(`Install ${pkg.filename} taken from ${packageUrl}`);
+        }
+        return installCondaPackages(
+          prefix,
+          packageUrl,
+          Module.FS,
+          untarjs,
+          verbose
         );
-      }
-      FS.writeFile(path, JSON.stringify(pkgCondaMeta));
-    } catch (error) {
-      console.error(error);
+      })
+    );
+    await waitRunDependencies(Module);
+    if (!skipLoadingSharedLibs) {
+      loadShareLibs(packages, sharedLibs, prefix, Module);
     }
   }
+  return packagesData;
+};
+
+const loadShareLibs = (
+  packages: IEmpackEnvMetaPkg[],
+  sharedLibs: FilesData[],
+  prefix: string,
+  Module: any
+) => {
+  console.log('sharedLibs', sharedLibs);
+  packages.map((pkg, i) => {
+    let packageShareLibs = sharedLibs[i];
+    if (Object.keys(packageShareLibs).length) {
+      loadDynlibsFromPackage(prefix, pkg.name, false, packageShareLibs, Module);
```

Comment:
You're not `await`ing this call, but [pyjs does it](https://github.com/emscripten-forge/pyjs/blob/0213cb4250d1ec9ce3a4225bfa4b4ca507fd15fd/include/pyjs/pre_js/load_pkg.js#L306)